app/push/server/push.js

```diff
@@ -115,44 +115,44 @@ export class PushClass {
 			data.headers.Authorization = this.options.getAuthorization();
 		}
 
-		return HTTP.post(`${ gateway }/push/${ service }/send`, data, (error, response) => {
-			if (response?.statusCode === 406) {
-				logger.info('removing push token', token);
-				appTokensCollection.remove({
-					$or: [{
-						'token.apn': token,
-					}, {
-						'token.gcm': token,
-					}],
-				});
-				return;
-			}
+		const response = await HTTP.post(`${ gateway }/push/${ service }/send`, data);
```

Comment:
don't we need error handling on this one? 👀 i don't know if push gateways errors will be handled by meteor or we need to handle them.

Reply:
this function keeps calling itself  up to 4 times if the status code from `response` is not 200 ([here](https://github.com/RocketChat/Rocket.Chat/blob/63587b414cc74bf3b17fcaeabea284b9b1dd2167/app/push/server/push.js#L142) and [here](https://github.com/RocketChat/Rocket.Chat/blob/63587b414cc74bf3b17fcaeabea284b9b1dd2167/app/push/server/push.js#L148))

i think that will catch any errors derived from the request

Reply:
Yeah but, if the package itself throws an error (not an error status code) won't that break the retries too:? 👀

Reply:
it's now in a try catch block :nerd_face: